### PR TITLE
Support Running local-up-cluster in CI

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

We have a new CI job for running conformance tests with local-up-cluster.sh: 
https://k8s-testgrid.appspot.com/sig-testing-misc#ci-kubernetes-local-e2e

In this PR, we do some additional tasks to be able to setup and run kubetest

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -947,6 +947,15 @@ EOF
 fi
 }
 
+# If we are running in the CI, we need a few more things before we can start
+if [ "$KUBETEST_IN_DOCKER" == "true" ]; then
```
Comment (cblecker) [inline hack/local-up-cluster.sh]: `if [[ "${KUBETEST_IN_DOCKER:-}" == "true" ]]; then`

File: hack/local-up-cluster.sh [hunk 2]
```diff
@@ -947,6 +947,15 @@ EOF
...
+  echo "Preparing to test ..."
+  $KUBE_ROOT/hack/install-etcd.sh
```
Comment (cblecker) [inline hack/local-up-cluster.sh]: `"${KUBE_ROOT}/hack/install-etcd.sh"`

File: hack/local-up-cluster.sh [hunk 3]
```diff
@@ -947,6 +947,15 @@ EOF
...
+  export PATH=$KUBE_ROOT/third_party/etcd:${PATH}
```
Comment (cblecker) [inline hack/local-up-cluster.sh]: `export PATH="${KUBE_ROOT}/third_party/etcd:${PATH}"`

File: hack/local-up-cluster.sh [hunk 4]
```diff
@@ -1022,3 +1031,10 @@ print_success
 if [[ "${ENABLE_DAEMON}" = false ]]; then
   while true; do sleep 1; done
 fi
+
+if [ "$KUBETEST_IN_DOCKER" == "true" ]; then
```
Comment (cblecker) [inline hack/local-up-cluster.sh]: `if [[ "${KUBETEST_IN_DOCKER:-}" == "true" ]]; then`

Comment (dims) [reply hack/local-up-cluster.sh]: Done!

File: hack/local-up-cluster.sh (hunk 3 above)
Comment (dims) [reply hack/local-up-cluster.sh]: Done!

File: hack/local-up-cluster.sh (hunk 2 above)
Comment (dims) [reply hack/local-up-cluster.sh]: Done!

File: hack/local-up-cluster.sh (hunk 1 above)
Comment (dims) [reply hack/local-up-cluster.sh]: Done!

File: hack/local-up-cluster.sh [hunk 5]
```diff
@@ -947,6 +947,15 @@ EOF
 fi
 }
 
+# If we are running in the CI, we need a few more things before we can start
+if [[ "${KUBETEST_IN_DOCKER:-}" == "true" ]]; then
+  echo "Preparing to test ..."
+  ${KUBE_ROOT}/hack/install-etcd.sh
+  export PATH="${KUBE_ROOT}/third_party/etcd:${PATH}"
+  KUBE_FASTBUILD=true make ginkgo cross
```
Comment (sttts) [inline hack/local-up-cluster.sh]: why cross compiling?

File: hack/local-up-cluster.sh [hunk 6]
```diff
@@ -947,6 +947,15 @@ EOF
 fi
 }
 
+# If we are running in the CI, we need a few more things before we can start
+if [[ "${KUBETEST_IN_DOCKER:-}" == "true" ]]; then
```
Comment (sttts) [inline hack/local-up-cluster.sh]: KUBETEST_IN_DOCKER does not say what it is. Maybe RUN_IN_KUBE_CI ?

File: hack/local-up-cluster.sh (hunk 5 above)
Comment (dims) [reply hack/local-up-cluster.sh]: the CI system is basically running kubetest with the "local" deployment option (https://github.com/kubernetes/test-infra/blob/master/kubetest/local.go) and it needs the `make cross` for the end-to-end conformance test to work properly.

File: hack/local-up-cluster.sh (hunk 6 above)
Comment (dims) [reply hack/local-up-cluster.sh]: @sttts KUBETEST_IN_DOCKER is already used by the CI system. please see https://storage.googleapis.com/kubernetes-jenkins/logs/ci-kubernetes-local-e2e/219/build-log.txt

**Pull request merged**